dining/views.py

```diff
@@ -600,6 +605,31 @@ def form_invalid(self, form):
         return HttpResponseRedirect(self.get_success_url())
 
     def form_valid(self, form):
-        form.send_reminder(self.request)
-        messages.success(self.request, "Diners have been informed")
-        return super(SlotPaymentView, self).form_valid(form)
+        # We use a critical section to prevent multiple emails from being sent.
```

Comment:
Arguably the logic below should be in the form as well instead of the view. Retrieve the error thrown by the form and communicate the message displayed based on the error-code the form has. This would be more in line with the division that Django made for view and form logic. That way we can also 'call' the form elsewhere if at any point things are moved around.

Also, it would be nice if you could write a test-case (I've really learned the importance of testcases over the lockdown) to catch the required behavior.